test_async.py

```diff
@@ -0,0 +1,79 @@
+import asyncio
+import aiohttp
+
+import mwapi
+from mwapi.errors import APIError
+
+async def query():
+    async with aiohttp.ClientSession() as s:
+        session = mwapi.AsyncSession('https://en.wikipedia.org', session=s)        
+        response = await asyncio.create_task(session.get(action='query', prop='revisions', revids=32423425))
+    print(response)
+    return response
+
+def single_query():
+    session = mwapi.Session('https://en.wikipedia.org')     
+    response = session.get(action='query', prop='revisions', revids=32423425)
+    print(response)
+    return response
+
+
+async def query_with_continuation():
+    async with aiohttp.ClientSession() as s:
+        session = mwapi.AsyncSession('https://en.wikipedia.org/', session=s)
+        continued = await asyncio.create_task(
+            session.get(
+                formatversion=2,
+                action='query',
+                generator='categorymembers',
+                gcmtitle='Category:17th-century classical composers',
+                gcmlimit=100,  # 100 results per request
+                continuation=True))
+        pages = []
+        try:
+            async for portion in continued:
+                if 'query' in portion:
+                    for page in portion['query']['pages']:
+                        pages.append(page['title'])
+                else:
+                    print("MediaWiki returned empty result batch.")
+        except APIError as error:
+            raise ValueError(
+                "MediaWiki returned an error:", str(error)
+            )
+    print("Fetched {} pages".format(len(pages)))
+    return pages
+
+def mwapi_query_with_continuation():
+    session = mwapi.Session('https://en.wikipedia.org/')
+    continued = session.get(
+        formatversion=2,
+        action='query',
+        generator='categorymembers',
+        gcmtitle='Category:17th-century classical composers',
+        gcmlimit=100,  # 100 results per request
+        continuation=True)
+
+    pages = []
+    try:
+        for portion in continued:
+            if 'query' in portion:
+                for page in portion['query']['pages']:
+                    pages.append(page['title'])
+            else:
+                print("MediaWiki returned empty result batch.")
+    except APIError as error:
+        raise ValueError(
+            "MediaWiki returned an error:", str(error)
+        )
+    print("Fetched {} pages".format(len(pages)))
+    return pages
+
+
+r1 = asyncio.run(query())
```

Comment:
Yes, you're right! This script is not for unit testing. I think I should change the name to demo_async.py to avoid confusion. It looks like the mwapi itself doesn't have unit tests. The [demo_mwoauth.py](https://github.com/AikoChou/python-mwapi/blob/master/demo_mwoauth.py), [demo_queries.py](https://github.com/AikoChou/python-mwapi/blob/master/demo_queries.py), [demo_timeout.py](https://github.com/AikoChou/python-mwapi/blob/master/demo_timeout.py), [demo_warning.py](https://github.com/AikoChou/python-mwapi/blob/master/demo_warning.py)  scripts are more like integration tests, right?

Reply:
Maybe we can explicitly call them integration_tests.py or similar, so it will be clear what is the purpose of them (maybe commenting somewhere that it needs an internet connection etc..). Adding unit tests is probably overkill for what we are trying to do, so I am fine if we skip them.

The alternative is to add something like pytests, but you'll need also to add fixtures/mocks to prevent calls to the MW API etc..

Reply:
Thanks for your advice! I moved these sample queries to README.md as an example usage of async (correspond to the mwapi session example). I think we can skip unit tests for now to make the commit simpler. :)